libraries/state_history/log.cpp

```diff
@@ -377,31 +478,46 @@ bool state_history_traces_log::exists(bfs::path state_history_dir) {
 state_history_chain_state_log::state_history_chain_state_log(const state_history_config& config)
     : state_history_log("chain_state_history", config) {}
 
-chain::bytes state_history_chain_state_log::get_log_entry(block_num_type block_num) {
+std::shared_ptr<std::vector<char>> state_history_chain_state_log::get_log_entry(block_num_type block_num) {
+   auto get_entry_from_disk = [&] {
+      auto [ds, _] = catalog.ro_stream_for_block(block_num);
+      if (ds.remaining()) {
+         return std::make_shared<std::vector<char>>(state_history::zlib_decompress(ds));
+      }
 
-   auto [ds, _] = catalog.ro_stream_for_block(block_num);
-   if (ds.remaining()) {
-      return state_history::zlib_decompress(ds);
-   }
+      if (block_num < begin_block() || block_num >= end_block())
+         return std::make_shared<std::vector<char>>();
+      state_history_log_header header;
+      get_entry_header(block_num, header);
+      return std::make_shared<std::vector<char>>(state_history::zlib_decompress(read_log));
+   };
 
-   if (block_num < begin_block() || block_num >= end_block())
-      return {};
-   state_history_log_header header;
-   get_entry_header(block_num, header);
-   return state_history::zlib_decompress(read_log);
+   if (thr.joinable()) {
```

Comment:
Same comment as above. Doesn't seem like this lambda is needed. Just add the `if (thr.joinable()) {` to the start of the method.